persistence/sql/persister_client.go

```diff
@@ -53,11 +53,9 @@ func (p *Persister) UpdateClient(ctx context.Context, cl *client.Client) (err er
 			}
 			cl.Secret = string(h)
 		}
-		// set the internal primary key
-		cl.ID = o.ID
```

Comment:
Why can we remove this?

Reply:
It is basically not used. For cockroach, we can just use a [hash-sharded index](https://www.cockroachlabs.com/docs/stable/hash-sharded-indexes) to avoid hot-spots. Mysql, postgres, and sqlite are fine with sequential keys (mysql actually prefers them).